File: packages/emitter/testcases/common/expected/extensions/system/Overlaps.cs
namespace TestCases.common.extensions.system
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class Overlaps
    {
        public static void run()
        {
            var s = "hello";
            var a = global::System.MemoryExtensions.AsSpan(s);
            var b = global::System.MemoryExtensions.AsSpan(s, 1);
            int off = 0;
            var ok1 = global::System.MemoryExtensions.Overlaps(a, b);
            var ok2 = global::System.MemoryExtensions.Overlaps(a, b, out off);
            global::System.Console.WriteLine($"ok1: {(global::js.Globals.String(ok1))}");
            global::System.Console.WriteLine($"ok2: {(global::js.Globals.String(ok2))} off: {(global::js.Globals.String(off))}");
        }
    }
}

File: packages/emitter/testcases/common/extensions/system/Overlaps.ts
import type { ExtensionMethods as SystemExt } from "@tsonic/dotnet/System.js";
import { Console } from "@tsonic/dotnet/System.js";
import { asinterface } from "@tsonic/core/lang.js";
import { int } from "@tsonic/core/types.js";

type Ext<T> = SystemExt<T>;

export function run(): void {
  const s = "hello";

  const a = asinterface<Ext<string>>(s).AsSpan();
  const b = asinterface<Ext<string>>(s).AsSpan(1);

  let off: int = 0;

  const ok1 = a.Overlaps(b);
  const ok2 = a.Overlaps(b, off);

  Console.WriteLine(`ok1: ${ok1}`);
  Console.WriteLine(`ok2: ${ok2} off: ${off}`);
}
